app/src/main/java/com/example/concordia_campus_guide/Adapters/PlaceToSearchResultAdapter.java

```diff
@@ -0,0 +1,120 @@
+import android.widget.ArrayAdapter;
+import android.widget.Filter;
+import android.widget.Filterable;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+
+import com.example.concordia_campus_guide.Models.Place;
+import com.example.concordia_campus_guide.R;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PlaceToSearchResultAdapter extends ArrayAdapter<Place> implements Filterable {
+
+    Context c;
+    List<Place> places;
+    List<Place> originalPlaces;
+    CustomFilter cs;
+
+    private static class ViewHolder {
+        private TextView itemView;
+    }
+
+    public PlaceToSearchResultAdapter(Context context, int textViewResourceId, List<Place> items) {
+        super(context, textViewResourceId, items);
+        c = context;
+        places = items;
+        originalPlaces = items;
+    }
+
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+        final ViewHolder viewHolder;
+
+        if (convertView == null) {
+            convertView = LayoutInflater.from(this.getContext())
+                    .inflate(R.layout.list_item_layout, parent, false);
+
+            viewHolder = new ViewHolder();
+            viewHolder.itemView = (TextView) convertView.findViewById(R.id.textView);
+
+            convertView.setTag(viewHolder);
+
+        } else {
+            viewHolder = (ViewHolder) convertView.getTag();
+        }
+
+        Place item = getItem(position);
+        if (item!= null) {
+            // My layout has only one TextView
+            // do whatever you want with your string
+            viewHolder.itemView.setText(item.getDisplayName());
+        }
+
+        return convertView;
+    }
+
+    @Override
+    public Place getItem(int i){
+        return places.get(i);
+    }
+
+    @Override
+    public int getCount(){
+        return places.size();
+    }
+
+    @NonNull
+    @Override
+    public Filter getFilter() {
+        if(cs == null){
+            cs = new CustomFilter();
+        }
+        return cs;
+    }
+
+    class CustomFilter extends Filter{
+
+        @Override
+        protected FilterResults performFiltering(CharSequence constraint) {
+            FilterResults results = new FilterResults();
+
+            if(constraint != null && constraint.length()>0){
+                constraint = constraint.toString().toUpperCase();
+
+                ArrayList<Place> filters = new ArrayList();
+
+                for(int i = 0; i<originalPlaces.size(); i++){
+                    if(originalPlaces.get(i).getDisplayName().toUpperCase().contains(constraint)){
+                        filters.add(originalPlaces.get(i));
+                    }
+                }
+
+                results.count = filters.size();
+                results.values = filters;
+            }
+            else{
+                results.count = originalPlaces.size();
+                results.values = originalPlaces;
+            }
+
+            return results;
```

Comment:
Do you think  it is possible to split the logic of these method into smaller ones so we can write some units tests for it easier ?

Reply:
I don't see how we could split this method up, its the fitlering function, it only has one functionality

Reply:
Rereading the method now, I noticed what you mean. It is  good. I might have misunderstood the functionality of the method when I read it the first time.